packages/stark-core/src/modules/error-handler/handlers/stark-error-handler.ts

```diff
@@ -0,0 +1,52 @@
+import {ErrorHandler, Injectable, Injector} from "@angular/core";
+import {Store} from "@ngrx/store";
+import {StarkCoreApplicationState} from "../../../common/store";
+import {StarkUnhandledError} from "../actions";
+import {STARK_LOGGING_SERVICE, StarkLoggingService} from "../../logging/services";
+
+
+@Injectable()
+export class StarkErrorHandler implements ErrorHandler {
+
+	private _starkLoggingService: StarkLoggingService;
+	private _applicationStore: Store<StarkCoreApplicationState>;
+
+	public constructor(private injector: Injector) {
+	}
+
+	/**
+	 * Thie method will dispatch an error method, which the user can then handle 
+	 * @param error the encountered error
+	 */
+	public handleError(error: any): void {
+		this.starkLoggingService.error("StarkErrorHandler: an error has occurred : ", error);
+		this.applicationStore.dispatch(new StarkUnhandledError(error));
+	}
+
+	/**
+	 * Gets the StarkLoggingService from the Injector (this is tried only once).
+	 * It returns 'undefined' if the service is not found (the Logging module is not imported in the app).
+	 */
+	private get starkLoggingService(): StarkLoggingService {
+		if (typeof this._starkLoggingService === "undefined") {
+			this._starkLoggingService = this.injector.get<StarkLoggingService>(STARK_LOGGING_SERVICE);
+			return this._starkLoggingService;
+		}
+
+		return this._starkLoggingService;
+	}
+
+	/**
+	 * Gets the Application Store from the Injector (this is tried only once).
+	 * It returns 'undefined' if the service is not found (the Logging module is not imported in the app).
+	 */
```

Comment:
In this case, this JSDoc is not accurate since the use case is a bit different, I suggest putting the following instead:

```
Gets the Application Store from the Injector.
@throws When the Store is not found (the NGRX Store module is not imported in the app).
```